control-plane/subcommand/fetch-server-region/command.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fetchserverregion
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/consul-k8s/control-plane/subcommand/common"
+	"github.com/hashicorp/consul-k8s/control-plane/subcommand/flags"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	defaultACLTokenFile = "/consul/login/acl-token"
+)
+
+// The consul-logout command issues a Consul logout API request to delete an ACL token.
+type Command struct {
+	UI cli.Ui
+
+	flagLogLevel string
+	flagLogJSON  bool
+	flagNodeName  string
+
+	flagSet *flag.FlagSet
+	k8s     *flags.K8SFlags
+
+	once   sync.Once
+	help   string
+	logger hclog.Logger
+
+	// for testing
+	clientset   kubernetes.Interface
+}
+
+type Locality struct {
+	Region string `json:"region"`
+}
+
+type Config struct {
+	Locality Locality `json:"locality"`
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagLogLevel, "log-level", "info",
+		"Log verbosity level. Supported values (in order of detail) are \"trace\", "+
+			"\"debug\", \"info\", \"warn\", and \"error\".")
+	c.flagSet.BoolVar(&c.flagLogJSON, "log-json", false,
+		"Enable or disable JSON output format for logging.")
+	c.flagSet.StringVar(&c.flagNodeName, "node-name", "",
+		"Specifies the node name that will be used.")
+
+	c.k8s = &flags.K8SFlags{}
+	flags.Merge(c.flagSet, c.k8s.Flags())
+
+	c.help = flags.Usage(help, c.flagSet)
+
+}
+
+func (c *Command) Run(args []string) int {
+	var err error
+	c.once.Do(c.init)
+
+	if err := c.flagSet.Parse(args); err != nil {
+		return 1
+	}
+
+	if c.logger == nil {
+		c.logger, err = common.Logger(c.flagLogLevel, c.flagLogJSON)
+		if err != nil {
+			c.UI.Error(err.Error())
+			return 1
+		}
+	}
+
+	if c.clientset == nil {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			// This just allows us to test it locally.
+			kubeconfig := clientcmd.RecommendedHomeFile
+			config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+			if err != nil {
+				c.UI.Error(err.Error())
+				return 1
+			}
+		}
+
+		c.clientset, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			c.UI.Error(err.Error())
+			return 1
+		}
+	}
+
+	if c.flagNodeName == "" {
+		c.UI.Error("-node-name is required")
+		return 1
+	}
+
+	config := c.fetchLocalityConfig()
+
+	jsonData, err := json.Marshal(config)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	fmt.Println(string(jsonData))
```

Comment:
This subcommand could take a file path as a flag and write the contents of the config to that file too. It is a pattern we have used in other subcommands as well.